tests/PHPStan/Analyser/data/bug-7385.php

```diff
@@ -0,0 +1,24 @@
+<?php declare(strict_types = 1);
+
+namespace Bug7385;
+
+use function PHPStan\Testing\assertType;
+
+class Model
+{
+	public function assertHasIface(): void
+	{
+		throw new \Error('For static analysis only, $this type is narrowed by MethodTypeSpecifyingExtension');
+	}
+}
+
+interface Iface
+{
+}
+
+function () {
+	$m = random_int(0, 1) === 0 ? new Model() : new class() extends Model implements Iface {};
+	assertType('Bug7385\Model', $m);
+	$m->assertHasIface();
+	assertType('Bug7385\Iface&Bug7385\Model', $m);
```

Comment:
@ondrejmirtes this assertion is not passing, do you know why?

- `Bug7385MethodTypeSpecifyingExtension::specifyTypes()` is correctly reached and `$type` is (the expected) `Bug7385\Iface&Bug7385\Model`
- I am not completely sure what `TypeSpecifierContext::createXxx()` is used for or if I should use `TypeSpecifierContext::createTrue()`, but I tried it.
- I also tried to to set `$overwrite` param to `true` in `$this->typeSpecifier->create()` factory method, but it seems the `$m` type is still `Bug7385\Model` as if the `MethodTypeSpecifyingExtension` result is completely ignored

Reply:
@ondrejmirtes can you please take a look on this issue?

Reply:
There's 343 open bug reports (https://github.com/phpstan/phpstan/labels/bug) and 323 feature requests (https://github.com/phpstan/phpstan/labels/feature-request). Your issue isn't anything urgent to warrant immediate attention - it's not anything special and doesn't stand out from other 666 open issues.

This PR doesn't even have an associated issue - I closed #7385 because there wasn't anything about type-specifying extensions in the description.

There might even be an open issue about "type-specifying extension doesn't allow me to change the type of the object the method is called on", I'm not sure.

Anyway, if you're already this far, you can take one more step and debug where it actually gets lost, and fix it.

Reply:
yes, there is https://github.com/phpstan/phpstan/issues/7396 issue I opened later after a short discussion about an implicit support in https://github.com/phpstan/phpstan/issues/7385